library/core/src/num/f32.rs

```diff
@@ -620,6 +620,106 @@ impl f32 {
         self.to_bits() & 0x8000_0000 != 0
     }
 
+    /// Returns the least number greater than `self`.
+    ///
+    /// Let `TINY` be the smallest representable positive `f32`. Then,
+    ///  - if `self.is_nan()`, this returns `self`;
+    ///  - if `self` is [`NEG_INFINITY`], this returns [`MIN`];
+    ///  - if `self` is `-TINY`, this returns -0.0;
+    ///  - if `self` is -0.0 or +0.0, this returns `TINY`;
+    ///  - if `self` is [`MAX`] or [`INFINITY`], this returns [`INFINITY`];
+    ///  - otherwise the unique least value greater than `self` is returned.
+    ///
+    /// The identity `x.next_up() == -(-x).next_down()` holds for all `x`. When `x`
```

Comment:
This is technically not true for `x = NAN`.

Reply:
Why not?

Reply:
Oh I think I see what you're getting at, `==` here is to be understood as a bitwise / definition equality, not floating point comparison where `NaN != NaN`.

Reply:
Should I update this doc comment to say "for all non-NaN `x`"? Or maybe reword it to "`x.next_up()` is equivalent to `-(-x).next_down()` for all `x`" to prevent confusion with `==` failing for NaNs?